### PR TITLE
remove unused variables + update eslint config to prevent them in future

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested *I guess...in that linting is passing.*
  - [x] Changes have been manually tested

#### What are the relevant tickets?
None; done toward #1061 because I was having a hard time telling how often some libraries were actually being used. (E.g.,`moment` was imported in several places where it was unused.)

#### What's this PR do?
Updates eslint config to forbid unused variables and removes current usage of unused variables.

**No functional changes.**

#### How should this be manually tested?
1. Add an unused variable.
2. Get a linting error.

### CODE REVIEW

File: frontend/public/src/containers/pages/DashboardPage_test.js [hunk 1]
```diff
@@ -1,29 +1,22 @@
...
 
 describe("DashboardPage", () => {
-  let helper, renderPage, userEnrollments, currentUser, isLinkableStub
+  let helper, renderPage, userEnrollments, currentUser
 
   beforeEach(() => {
     helper = new IntegrationTestHelper()
     userEnrollments = [makeCourseRunEnrollment(), makeCourseRunEnrollment()]
     currentUser = makeUser()
-    isLinkableStub = helper.sandbox.stub(courseApi, "isLinkableCourseRun")
```
Comment (ChristopherChudzicki) [inline frontend/public/src/containers/pages/DashboardPage_test.js:26]: The only things removed in this PR are
- unused variable declarations
- unused imports
- assignments to variables which were never ready, as in this case.

**Comment by pdpinch** [comment]
Where does our eslint file come from? is it in the cookie cutter? Does it need to be updated there too?

**Comment by ChristopherChudzicki** [comment]
@pdpinch Most/all of our projects extend from https://github.com/mitodl/eslint-config, which does specify a `"no-unused-vars"` rule, but this project was specifically turning it off with `"no-unused-vars": 0`. _The override was specified in the first PR in this repo, https://github.com/mitodl/mitxonline/pull/6 ... I'm guessing as a temporary measure that we forgot to remove. @rhysyngsun maybe knows_

In general, though, the shared config https://github.com/mitodl/eslint-config should definitely be updated for a variety of reasons (largely to due with typescript). It's an npm package, so it can be updated without breaking existing repo linting.

**Comment by rhysyngsun** [comment]
> @pdpinch Most/all of our projects extend from https://github.com/mitodl/eslint-config, which does specify a `"no-unused-vars"` rule, but this project was specifically turning it off with `"no-unused-vars": 0`. _The override was specified in the first PR in this repo, #6 ... I'm guessing as a temporary measure that we forgot to remove. @rhysyngsun maybe knows_
> 
> In general, though, the shared config https://github.com/mitodl/eslint-config should definitely be updated for a variety of reasons (largely to due with typescript). It's an npm package, so it can be updated without breaking existing repo linting. I can start a discussion.

No idea, it was Alice who setup the eslint stuff.

**Review by collinpreston — APPROVED** [review]
Works for me!

**Pull request merged**